graph/src/data/schema.rs

```diff
@@ -73,8 +74,204 @@ pub enum SchemaValidationError {
         _0, _1
     )]
     ImportedTypeUndefined(String, String), // (type_name, schema)
+    #[fail(display = "Fulltext directive name undefined")]
+    FulltextNameUndefined,
+    #[fail(display = "Fulltext directive name overlaps with type: {}", _0)]
+    FulltextNameConflict(String),
+    #[fail(
+        display = "Fulltext directive name overlaps with existing entity field: {}",
+        _0
+    )]
+    FulltextNameCollision(String),
+    #[fail(display = "Fulltext language is undefined")]
+    FulltextLanguageUndefined,
+    #[fail(display = "Fulltext language is invalid: {}", _0)]
+    FulltextLanguageInvalid(String),
+    #[fail(display = "Fulltext algorithm is undefined")]
+    FulltextAlgorithmUndefined,
+    #[fail(display = "Fulltext algorithm is invalid: {}", _0)]
+    FulltextAlgorithmInvalid(String),
+    #[fail(display = "Fulltext include is invalid")]
+    FulltextIncludeInvalid,
+    #[fail(display = "Fulltext directive requires an 'include' list")]
+    FulltextIncludeUndefined,
+    #[fail(display = "Fulltext 'include' list must contain an object")]
+    FulltextIncludeObjectMissing,
+    #[fail(
+        display = "Fulltext 'include' object must contain 'entity' (String) and 'fields' (List) attributes"
+    )]
+    FulltextIncludeEntityMissingOrIncorrectAttributes,
+    #[fail(display = "Fulltext directive includes an entity not found on the subgraph schema")]
+    FulltextIncludedEntityNotFound,
+    #[fail(display = "Fulltext include field must have a 'name' attribute")]
+    FulltextIncludedFieldMissingRequiredProperty,
+    #[fail(
+        display = "Fulltext included Entity field, {}, not found or not a string",
```

Comment:
Perhaps we can shorten this message to `Fulltext entity field `{}` not found or not a string`?